### PR TITLE
[CIS-1009] Fix the appearance of system messages

### PR DESCRIPTION
**Pull request by SagarSDagdu**
### What does this PR do
- This PR fixes the appearance of the system messages in the channel

Before             |  After
:-------------------------:|:-------------------------:
 |

### CODE REVIEW

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView_Tests.swift [hunk 1]
```diff
@@ -93,6 +93,30 @@ final class ChatMessageContentView_Tests: XCTestCase {
...
     }
+    
+    func test_appearanceForSystemMessage() {
+        // Create a system message
+        let systemMessage: ChatMessage = .mock(
+            id: .unique,
+            cid: .unique,
+            text: "Some member was added",
+            type: .system,
+            author: myFriend,
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView_Tests.swift:104]: out of curiosity, who is set as an author for a system message? 🤔 I'd expect the `author == nil` but the field in non-optional

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift [hunk 2]
```diff
@@ -39,6 +39,11 @@ open class _ChatMessageLayoutOptionsResolver<ExtraData: ExtraDataTypes> {
         
         var options: ChatMessageLayoutOptions = []
 
+        // The text should be centered without a bubble for system messages
+        guard message.type != .system else {
+            return [.text, .centered]
+        }
+
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift:46]: let's have it also covered by the unit-test

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift [hunk 3]
```diff
@@ -356,7 +356,15 @@ open class _ChatMessageContentView<ExtraData: ExtraDataTypes>: _View, ThemeProvi
...
         ].compactMap { $0 }
 
-        if options.contains(.flipped) {
+        if options.contains(.centered) {
+            mainContainer.addArrangedSubviews([bubbleThreadMetaContainer])
+            
+            constraintsToActivate += [
+                mainContainer.centerXAnchor
+                    .pin(equalTo: centerXAnchor)
+                    .almostRequired
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift:365]: why almost required? 🤔

**Review by evsaev — APPROVED** [review]
LGTM 👍

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift (hunk 2 above)
Comment (SagarSDagdu) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift:46]: Thanks! I've added a unit test for this

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift (hunk 3 above)
Comment (SagarSDagdu) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift:365]: This was not required, removed it

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView_Tests.swift (hunk 1 above)
Comment (SagarSDagdu) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView_Tests.swift:104]: The `author` of a system message is not nil. In this case, it is the creator of the channel

**Review by DominikBucher12 — APPROVED** [review]
Awesome 🚀

**Pull request merged**